### DISCUSSION TITLE
Is there a different input mapper that would work with Steam Deck?

### DISCUSSION THREAD

**Discussion by randomname42** [sezanzeb/input-remapper#923] [Q&A]
I was going to make a request for a flatpak version, as there doesn't seem to be a way to use this on Steam Deck (SteamOS is based off of Arch, but "yay -S input-remapper-git" gives an error "bash: yay: command not found". If there is a way to disable SteamOS's readonly state and set up a bunch of other stuff to get that working it would be well beyond my skill and knowledge level to do. Plus I think any changes like that would get erased every time SteamOS is updated, due to the immutable file system. That would be very impractical.)

But I see input-mapper isn't being developed anymore, so it sounds like a request for a flatpak would never happen.

Is there a different input mapper for Linux that would work with a Steam Deck? 

I've tried AntiMicroX but it doesn't let you remap input from one analog axis to a different analog stick, which is something I specifically need to do.

**Comment by Jahfry** [comment]
9 months later, but I just came across this thread. 

I solved this need on a Steam Deck using 'keyd'. It wasn't painless, and doesn't have a GUI, but if you're willing to go the extra mile [this](https://www.reddit.com/r/SteamDeck/comments/1dzbs14/guide_keyd_remap_those_usb_device_keys_both/) may help. 

Unfortunately there's not a universal solution. Looks like 'keyd' on SteamOS for now (and possibly having to redo it after system updates) and Input Mapper on things like Bazzite. 

With the way Wayland is starting to take hold I have hopes there will be a solution in the future that addresses all. 

Other tools to keep an eye on: Hawck, ydotool, Autokey, kmonad, AHK_X11 (but as far as I can see none are full easy solutions with flatpacks & working in Wayland in the same solution).
